bazel/src/main/java/com/google/api/codegen/bazel/BazelBuildFileView.java

```diff
@@ -43,6 +44,10 @@ class BazelBuildFileView {
     tokens.put("version", bp.getVersion());
     tokens.put("package", bp.getProtoPackage());
 
+    // For regeneration of Java rules, we are particularly interestd in what the saved transport value was,
```

Comment:
As above, not clear to me why we need a special path for Java as opposed to `bp.forceTransport`

Reply:
The PR description explains this but let me know where I can add more commentary.

Java is the only one (as of this PR) that needs extensive modification of _all_ of its rules based on the transport selected. So there is a special path for Java. This could change, say, for Go if we wanted forgo generation of gRPC stubs when `rest` only, but that's a separate PR imo, I'm addressing the bug related to Java for now.

Reply:
I think the naming is probably to blame here.  So the "override" comes from the buildozer "caching" of the existing file, whereas `forceTransport` comes from the CLI flag being provided. So the "override" is used when `forceTransport` is **not** set. I actually changed in ApiVersoinedDir the buildozer "caching" to ignore java_gapic_library `transport` attribute when the `forceTransport` (i.e. the flag) was set.

Did I answer the question? This is definitely some complex logic.

Reply:
Yes, it was the naming. I remember when I did work on this code, I found the `override` nomenclature misleading. Thanks for clarifying, here and in the code. Things are now a lot clearer!